docs/debugging.md

```diff
@@ -0,0 +1,94 @@
+**Table of contents**
+
+<!--- TOC -->
+
+* [Debugging coroutines](#debugging-coroutines)
+* [Debug mode](#debug-mode)
+* [Stacktrace recovery](#stacktrace-recovery)
+  * [Stacktrace recovery machinery](#stacktrace-recovery-machinery)
+* [Debug agent](#debug-agent)
+  * [Debug agent and Android](#debug-agent-and-android)
+
+<!--- END_TOC -->
+
+
+## Debugging coroutines
+Asynchronous programming is hard and debugging asynchronous programs is even harder.
```

Comment:
Scary statement :) I'd rather simply say "Debugging asynchronous programs is challenging, because multiple concurrent coroutines are typically working at the same time. To help with that, `kotlinx.coroutines` .... " (and so on)